test/unittests/utils/test_xml_validation_models.py

```diff
@@ -0,0 +1,374 @@
+import pandas as pd
+import pytest
+from lxml import etree
+from pandas.testing import assert_frame_equal
+
+from dsp_tools.utils.xml_validation_models import (
+    InconsistentTextValueEncodings,
+    TextValueData,
+    _check_only_one_valid_encoding_used_all_props,
+    _check_only_one_valid_encoding_used_one_prop,
+    _get_all_ids_prop_encoding_from_root,
+    _get_id_prop_encoding_from_one_resource,
+    _get_prop_encoding_from_one_property,
+    check_if_only_one_encoding_is_used_in_xml,
+)
+
+
+def test_get_prop_encoding_from_all_properties_no_text() -> None:
+    test_props = etree.fromstring(
+        """
+        <resource label="First Testthing"
+              restype=":TestThing"
+              id="test_thing_1"
+              permissions="res-default">
+            <uri-prop name=":hasUri">
+                <uri permissions="prop-default">https://dasch.swiss</uri>
+            </uri-prop>
+            <boolean-prop name=":hasBoolean">
+                <boolean permissions="prop-default">true</boolean>
+            </boolean-prop>
+        </resource>
+        """
+    )
+    res = _get_id_prop_encoding_from_one_resource(test_props)
+    assert res is None
```

Comment:
if you follow my recommendation to return an empty list instead of None, then you have to write:

```suggestion
    assert not res
```